datafusion/physical-optimizer/src/pruning.rs

```diff
@@ -1538,13 +1550,21 @@ fn build_predicate_expression(
             build_predicate_expression(&right, schema, required_columns, unhandled_hook);
         // simplify boolean expression if applicable
         let expr = match (&left_expr, op, &right_expr) {
+            (left, Operator::And, right)
```

Comment:
I think we should leave the and/or folding to a dedicated optimizer pass (IIRC that's "expression simplification") instead of re-implementing it here again. It's unnecessarily complex, potentially slow, requires more tests to write and maintain, and is a potential source of additional bugs.

Reply:
+1, I'll file an issue to record the task as a follow-up